src/routes/(app)/docs/[lang]/[filename]/+server.ts

```diff
@@ -0,0 +1,50 @@
+import { rootDir } from '$lib/server/root-dir';
+import { error } from '@sveltejs/kit';
+import fs from 'fs';
+import { join } from 'path';
+import { z } from 'zod';
+
+export async function GET({ params }) {
+	const docsDir = join(rootDir, 'docs');
+
+	const validFolderNameRegex = /^[a-z]{2}(-[a-z]{2})?$/i;
+	const availableLangs = fs
+		.readdirSync(docsDir)
+		.filter(
+			(folder) =>
+				validFolderNameRegex.test(folder) && fs.statSync(join(docsDir, folder)).isDirectory()
+		);
+
+	const langSchema = z.enum(availableLangs as [string, ...string[]]);
+	let lang: string;
+
+	try {
+		lang = langSchema.parse(params.lang);
+	} catch {
+		throw error(400, 'Invalid language');
+	}
+
+	const folderPath = join(docsDir, lang);
+
+	const availableFiles = fs
+		.readdirSync(folderPath)
+		.filter((file) => file.endsWith('.md') && fs.statSync(join(folderPath, file)).isFile());
+
+	const filenameSchema = z.enum(availableFiles as [string, ...string[]]);
+	let filename: string;
+
+	try {
+		filename = filenameSchema.parse(params.filename);
+	} catch {
+		throw error(404, 'File not found');
+	}
```

Comment:
I improved zod error handling in https://github.com/be-BOP-io-SA/be-BOP/pull/1782/commits/fa8fcd81ac0a7b23f630fa42b146bc63c7ecdc5e, you can just `parse` without error management